perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -639,6 +641,22 @@ public void updateUserExtSourceLastAccess(PerunSession sess, UserExtSource userE
 		return getUsersManagerImpl().getUserExtSources(sess, user);
 	}
 
+	@Override
+	public List<RichUserExtSource> getRichUserExtSources(PerunSession sess, User user, List<String> attrsNames) throws InternalErrorException {
+		return getUserExtSources(sess, user).stream()
+			.map(ues -> new RichUserExtSource(ues,
+				attrsNames == null ? getPerunBl().getAttributesManagerBl().getAttributes(sess, ues)
+					: getPerunBl().getAttributesManagerBl().getAttributes(sess, ues, attrsNames)))
+			.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<RichUserExtSource> filterOnlyAllowedAttributesForRichUserExtSources(PerunSession sess, List<RichUserExtSource> richUserExtSources) {
+		richUserExtSources.forEach(rues -> rues.setAttributes(
+			AuthzResolverBlImpl.filterNotAllowedAttributes(sess, rues.asUserExtSource(), rues.getAttributes())));
```

Comment:
It doesn't have to be part of this pull-request, but asking for authorization on all attributes can be really time consuming operation. We could "cache" results of authorization for one caller and all UserExtSources of single user. Like shared context, as we do in filtering attributes for RichMember and Groups. Because if you are allowed to read "cn" attribute of UserExtSource for one user, you have same authorization for the rest of his UserExtSources.